front/app/modules/commercial/insights/admin/containers/Insights/Details/Categories.tsx

```diff
@@ -0,0 +1,141 @@
+
+const Container = styled.div`
+  background-color: #fff;
+  padding: 28px;
+
+  h1 {
+    color: ${colors.adminTextColor};
+    font-size: ${fontSizes.large}px;
+    display: flex;
+    align-items: center;
+    button {
+      margin-left: 10px;
+    }
+  }
+`;
+
+const CategoriesContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  .categoryTag {
+    margin-right: 8px;
+  }
+`;
+
+const EmptyStateContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 16px 24px;
+  background-color: ${colors.clBlueLightest};
+  color: ${colors.adminTextColor};
+  border-radius: 3px;
+
+  .content {
+    width: 80%;
+  }
+
+  .title {
+    margin: 0;
+    padding: 0;
+    font-size: ${fontSizes.base}px;
+    font-weight: bold;
+  }
+`;
+
+const Categories = ({
+  location: { pathname, query },
+  params: { viewId },
+  intl: { formatMessage },
+}: CategoryProps) => {
+  const categories = useInsightsCategories(viewId);
+
+  if (isNilOrError(categories)) {
+    return null;
+  }
+
+  const handleCategoryClick = (id: string) => () => {
+    const category = query.category === id ? undefined : id;
+    clHistory.push({
+      pathname,
+      search: stringify({ ...query, category }, { addQueryPrefix: true }),
+    });
+  };
+
+  return (
+    <Container data-testid="insightsDetailsCategories">
+      <h1>
+        {formatMessage(messages.categoriesTitle)}
+        <IconTooltip content={formatMessage(messages.categoriesTitleTooltip)} />
+      </h1>
+      {categories.length > 0 ? (
+        <CategoriesContainer>
+          <div>
+            {categories.map((category) => (
+              <Tag
+                key={category.id}
+                label={category.attributes.name}
+                variant={
+                  query.category === category.id ? 'primary' : 'secondary'
+                }
+                count={category.attributes.inputs_count}
+                className="categoryTag"
+                onClick={handleCategoryClick(category.id)}
+              />
+            ))}
+          </div>
+          <Button buttonStyle="admin-dark" linkTo={`${pathname}/edit`}>
+            {formatMessage(messages.editCategories)}
+          </Button>
+        </CategoriesContainer>
+      ) : (
+        <EmptyStateContainer data-testid="insightsDetailsCategoriesEmpty">
+          <div className="content">
+            <p className="title">
+              {formatMessage(messages.categoriesEmptyTitle)}
+            </p>
+            <p> {formatMessage(messages.categoriesEmptyDescription)}</p>
+          </div>
+
+          <Button buttonStyle="admin-dark" linkTo={`${pathname}/edit`}>
+            {formatMessage(messages.categoriesEmptyButton)}
+          </Button>
+        </EmptyStateContainer>
+      )}
+    </Container>
+  );
+};
+
+export default withRouter(injectIntl(Categories));
+
+// categoriesSeeAll: {
```

Comment:
This is a strange place to declare a message :)

Reply:
Hahah, forgot to remove, thanks!